gunicorn/util.py

```diff
@@ -543,3 +543,14 @@ def app(environ, start_response):
         return [msg]
 
     return app
+
+
+def split_request_uri(uri):
+    if uri.startswith("//"):
+        # When the path starts with //, urlsplit considers it as a
+        # relative uri while the RFC says we should consider it as abs_path
+        # http://www.w3.org/Protocols/rfc2616/rfc2616-sec5.html#sec5.1.2
+        parts = _compat.urlsplit("." + uri)
```

Comment:
not sure i understand this line. What trigger the "." here ?

Reply:
I'm not sure the comment makes sense, but a quick test in the console shows that this forces `urlsplit` to consider the string as a path—with no scheme or netloc—rather than parsing this as a scheme-relative URL with the first path component as the netloc.

Reply:
"." forces urlsplit to parse path part as RFC says:
```python
In [5]: urlsplit('//path?query#fragment')
Out[5]: SplitResult(scheme='', netloc='path', path='', query='query', fragment='fragment')

In [6]: urlsplit('.//path?query#fragment')
Out[6]: SplitResult(scheme='', netloc='', path='.//path', query='query', fragment='fragment')
```

Reply:
Can you add a test case in `tests/test_util.py`? It's easy to introduce a regression when fixing another issue in the urllib module. This way we can know early if something has changed in the development version of Python.

Reply:
@tilgovi thanks. imo that worlrth a comment for readability.